### ISSUE TITLE
Problem with 'requests' dependency on CentOS 7

### ISSUE THREAD

**Issue by verdurin** [gc3-uzh-ch/elasticluster#441]
Just installed from GitHub on a CentOS 7 VM, and saw this error when trying to run `elasticluster`:
```
Successfully installed Babel-2.3.4 MarkupSafe-1.0 PrettyTable-0.7.2 PyCLI-2.0.3 PyJWT-1.5.0 PyYAML-3.12 adal-0.4.5 ansible-2.3.0.0 apache-libcloud-2.0.0 asn1crypto-0.22.0 azure-2.0.0 azure-batch-3.0.0 azure-common-1.1.6 azure-datalake-store-0.0.9 azure-graphrbac-0.30.0 azure-keyvault-0.3.3 azure-mgmt-1.0.0 azure-mgmt-authorization-0.30.0 azure-mgmt-batch-4.0.0 azure-mgmt-cdn-0.30.3 azure-mgmt-cognitiveservices-1.0.0 azure-mgmt-compute-1.0.0 azure-mgmt-containerregistry-0.2.1 azure-mgmt-datalake-analytics-0.1.4 azure-mgmt-datalake-nspkg-2.0.0 azure-mgmt-datalake-store-0.1.4 azure-mgmt-devtestlabs-2.0.0 azure-mgmt-dns-1.0.1 azure-mgmt-documentdb-0.1.3 azure-mgmt-iothub-0.2.2 azure-mgmt-keyvault-0.31.0 azure-mgmt-logic-2.1.0 azure-mgmt-monitor-0.2.1 azure-mgmt-network-1.0.0 azure-mgmt-nspkg-2.0.0 azure-mgmt-rdbms-0.1.0 azure-mgmt-redis-4.1.0 azure-mgmt-resource-1.1.0 azure-mgmt-scheduler-1.1.2 azure-mgmt-sql-0.5.1 azure-mgmt-storage-1.0.0 azure-mgmt-trafficmanager-0.30.0 azure-mgmt-web-0.32.0 azure-nspkg-2.0.0 azure-servicebus-0.21.1 azure-servicefabric-5.6.130 azure-servicemanagement-legacy-0.20.6 azure-storage-0.34.2 boto-2.47.0 certifi-2017.4.17 cffi-1.10.0 chardet-3.0.3 click-6.7 cliff-2.7.0 cmd2-0.7.2 coloredlogs-7.0 cryptography-1.9 debtcollector-1.14.0 elasticluster enum34-1.1.6 funcsigs-1.0.2 functools32-3.2.3.post2 futures-3.1.1 google-api-python-client-1.6.2 google-compute-engine-2.3.7 httplib2-0.10.3 humanfriendly-3.2 idna-2.5 ipaddress-1.0.18 iso8601-0.1.11 isodate-0.5.4 jinja2-2.9.6 jsonpatch-1.15 jsonpointer-1.10 jsonschema-2.6.0 keyring-10.3.2 keystoneauth1-2.20.0 monotonic-1.3 msgpack-python-0.4.8 msrest-0.4.8 msrestazure-0.4.7 netaddr-0.7.19 netifaces-0.10.5 oauth2client-4.1.0 oauthlib-2.0.2 os-client-config-1.27.0 osc-lib-1.6.0 oslo.config-4.2.0 oslo.i18n-3.15.3 oslo.serialization-2.18.0 oslo.utils-3.25.1 paramiko-2.1.2 pathlib2-2.2.1 pbr-3.0.1 positional-1.1.1 pyasn1-0.2.3 pyasn1-modules-0.0.8 pycparser-2.17 pycrypto-2.6.1 python-cinderclient-2.1.0 python-dateutil-2.6.0 python-gflags-3.1.1 python-glanceclient-2.6.0 python-keystoneclient-3.10.0 python-neutronclient-6.3.0 python-novaclient-8.0.0 pytz-2017.2 requests-2.17.3 requests-oauthlib-0.8.0 requestsexceptions-1.2.0 rfc3986-1.0.0 rsa-3.4.2 scandir-1.5 schema-0.6.6 secretstorage-2.3.1 simplejson-3.10.0 stevedore-1.22.0 unicodecsv-0.14.1 uritemplate-3.0.0 urllib3-1.21.1 warlock-1.2.0 wrapt-1.10.10
(elasticluster)[centos@provtest7 src]$ elasticluster
Traceback (most recent call last):
  File "/home/centos/src/elasticluster/bin/elasticluster", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/home/centos/src/elasticluster/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3036, in <module>
    @_call_aside
  File "/home/centos/src/elasticluster/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3020, in _call_aside
    f(*args, **kwargs)
  File "/home/centos/src/elasticluster/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3049, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/home/centos/src/elasticluster/lib/python2.7/site-packages/pkg_resources/__init__.py", line 656, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/home/centos/src/elasticluster/lib/python2.7/site-packages/pkg_resources/__init__.py", line 669, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/home/centos/src/elasticluster/lib/python2.7/site-packages/pkg_resources/__init__.py", line 859, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (requests 2.17.3 (/home/centos/src/elasticluster/lib/python2.7/site-packages), Requirement.parse('requests~=2.14.1'), set(['msrest']))
```

**Comment by A1ve5** [comment]
Same issue on Ubuntu 16.04.2 LTS.
Downgrading requests as suggested fixed the problem.

**Comment by riccardomurri** [comment]
Fixed by PR #444 -- many thanks to @verdurin for contributing the patch alongside with the report!
